java-shared-dependencies/dependency-analyzer/src/main/java/com/google/cloud/model/AnalyzeReport.java

```diff
@@ -0,0 +1,99 @@
+package com.google.cloud.model;
+
+import com.google.common.collect.ImmutableSet;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+public class AnalyzeReport {
+
+  private final VersionKey root;
+  private final Map<VersionKey, List<Advisory>> advisories;
+  private final Map<VersionKey, List<String>> nonCompliantLicenses;
+  private final ImmutableSet<LicenseCategory> compliantCategories = ImmutableSet.of(
+      LicenseCategory.NOTICE);
+
+  private final static Logger LOGGER = Logger.getLogger(AnalyzeReport.class.getName());
+
+  public AnalyzeReport(VersionKey root, List<PackageInfo> result) {
+    this.root = root;
+    advisories = getAdvisories(result);
+    nonCompliantLicenses = getNonCompliantLicenses(result);
+  }
+
+  public AnalyzeResult generateReport() {
```

Comment:
generateReport should return a "report", not "result".

Reply:
The name is a bit confusing.

I think this should be the English sentence to describe the process:
The analyzer analyzes a package and returns a analysis result.
The analysis result generated a analysis report containing risk information and returns PASS or FAIL.

WDYT?